src/plugins/discover/public/components/data_types/logs/log_level_badge_cell.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiBadge, mathWithUnits, useEuiTheme } from '@elastic/eui';
+import type { CSSObject } from '@emotion/react';
+import { getLogLevelCoalescedValue, getLogLevelColor } from '@kbn/discover-utils';
+import { euiThemeVars } from '@kbn/ui-theme';
+import type { DataGridCellValueElementProps } from '@kbn/unified-data-table';
+import React from 'react';
+
+const badgeCss: CSSObject = {
+  marginTop: '-4px',
+  maxWidth: mathWithUnits(euiThemeVars.euiSize, (size) => size * 7.5),
+};
+
+export const getLogLevelBadgeCell =
+  (logLevelField: string) => (props: DataGridCellValueElementProps) => {
+    const { euiTheme } = useEuiTheme();
+    const value = props.row.flattened[logLevelField];
+
+    if (!value) {
+      return <span data-test-subj="logLevelBadgeCell-empty">-</span>;
+    }
+
+    const coalescedValue = getLogLevelCoalescedValue(value);
+    const color = coalescedValue ? getLogLevelColor(coalescedValue, euiTheme) : undefined;
+
+    return (
+      <EuiBadge
+        color={color}
+        data-test-subj={`logLevelBadgeCell-${coalescedValue ?? 'unknown'}`}
+        css={badgeCss}
+      >
+        {value}
```

Comment:
I think for some badges we might want to use white text instead of dark. For example for `emerg` badge here:

<img width="642" alt="Screenshot 2024-07-16 at 16 18 35" src="https://github.com/user-attachments/assets/786ae648-f48a-4fea-b34a-f0866d43b83e">

Or is it fine with [the default dark color text](https://github.com/elastic/eui/blob/1dd68eb8d8798d6a476bc7b0757979276d6ea182/packages/eui/src/components/badge/badge.tsx#L143) regardless of current badge background color?

cc @MichaelMarcialis

Reply:
I merged this PR to get the feature in, but we can make followup changes around this if needed based on @MichaelMarcialis' feedback.

Reply:
Thanks for the ping, @jughosta and @davismcphee! Yes, the badge text color used in the mockups (#000000) is correct and passes the AA WCAG accessibility standard for normal text. Switching to white will actually cause the text within the badge to fail this contrast check, even in the `emerg` badge.

Reply:
Thanks for confirming @MichaelMarcialis!